Db.php

```diff
@@ -194,14 +284,20 @@ static function expandRefs($value) {
      *   skip - the number of objects to skip
      *
      * @param string $collection
-     * @param array $query
-     * @param array $options
+     * @param array  $query
+     * @param array  $options
+     *
      * @return MongoCursor
      **/
-    static function find($collection, $query = array(), $options = array()) {
-        $col = self::getCollection($collection, true);
-        $fields = isset($options['fields']) ? $options['fields'] : array();
-        $result = $col->find($query, $fields);
+    public function find($collection, $query = array(), $options = array())
+    {
+        $this->readOnly = true;
```

Comment:
I don't think we want a `find` method to have stateful side-effects -- the `readOnly` state should only be set if you explicitly want to tell it that writes are not permitted.